### DISCUSSION TITLE
Renaming targets with dagster-airbyte

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17774] [Q&A] [answered]
Hi all,

I've started using Dagster along with dagster-airbyte some time ago.

Basing on my current use case I am wondering if it is possible to define or map target names of the streams (i.e. tables names in target database) when using `load_assets_from_airbyte_instance` as follows:

```
airbyte_assets = load_assets_from_airbyte_instance(
    airbyte_instance,
    key_prefix="...",
    connection_to_group_fn=connection_to_group
    )
```
	
I am getting following error when using it:

```
dagster._check.CheckError: Failure condition: Invalid out key 'sdgggdf-activity-gfdg-trt' supplied to `internal_asset_deps` argument for multi-asset airbyte_sync_0882c. Must be one of the outs for this multi-asset ['sdgggdf_activity_gfdg_trt', '...']	
```
The reason is that in source database (dynamodb) the names of tables have "-" in their names.

I would like to map and rename the target names and replace "-" by "_".

Is it currently possible? Or is it planned to be implemented/deployed in nearest future?
I've tried to find it in the docu but I do not think is there.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C047VPGQUTV/p1699368757969899?thread_ts=1699368757.969899&cid=C047VPGQUTV).*

**Comment by jamiedemaria** [comment, marked as answer]
`load_assets_from_airbyte_instance` has a parameter `connection_to_asset_key_fn` that you can use to map a connection and table name to an asset key 

https://docs.dagster.io/_apidocs/libraries/dagster-airbyte#dagster_airbyte.load_assets_from_airbyte_instance

**Reply by WojciechSzybaVeVe** [reply]
Using this function as follows: `connection_to_asset_key_fn=lambda c, n: AssetKey([c.name, n.replace('-', '_')])` ends up with the same error because the stream names in Airbyte derived from source database are still the same (contains "-" instead of "_")

**Reply by jamiedemaria** [reply]
Ok - sounds like something we'll need to address on our side. Can you open a github issue for this? I'll get it directed to the right person

**Reply by jamiedemaria** [reply]
issue - https://github.com/dagster-io/dagster/pull/18474

**Comment by rtesselli** [comment]
Following because I have the same issue
